python-checks/src/main/java/org/sonar/python/checks/cdk/DisabledESDomainEncryptionCheck.java

```diff
@@ -0,0 +1,118 @@
+      checkDomain("encryption_at_rest_options", "aws_cdk.aws_opensearchservice.CfnDomain.EncryptionAtRestOptionsProperty", OPENSEARCH));
+    checkFqn("aws_cdk.aws_elasticsearch.Domain",
+      checkDomain("encryption_at_rest", "aws_cdk.aws_elasticsearch.EncryptionAtRestOptions", ELASTICSEARCH));
+    checkFqn("aws_cdk.aws_elasticsearch.CfnDomain",
+      checkDomain("encryption_at_rest_options", "aws_cdk.aws_elasticsearch.CfnDomain.EncryptionAtRestOptionsProperty", ELASTICSEARCH));
+  }
+
+  private static BiConsumer<SubscriptionContext, CallExpression> checkDomain(String encryptionArgName, String argFqnMethod, String engine) {
+    return (ctx, callExpression) -> getArgument(ctx, callExpression, encryptionArgName)
+      .ifPresentOrElse(
+        argEncryptedTrace -> argEncryptedTrace.addIssueIf(isSensitiveOptionObj(ctx, argFqnMethod).or(isSensitiveOptionDict(ctx)), unencryptedMessage(engine)),
+        () -> ctx.addIssue(callExpression.callee(), omittingMessage(encryptionArgName, engine))
+      );
+  }
+
+  private static  String omittingMessage(String encryptionArgName, String engine) {
+    return String.format(OMITTING_MESSAGE, encryptionArgName, engine);
+  }
+
+  private static String unencryptedMessage(String engine) {
+    return String.format(UNENCRYPTED_MESSAGE, engine);
+  }
+
+  private static Predicate<Expression> isSensitiveOptionObj(SubscriptionContext ctx, String argFqnMethod) {
+    return expr -> {
+      if (!isFqn(argFqnMethod).test(expr)) {
+        return false;
+      }
+      if (!expr.is(Tree.Kind.CALL_EXPR)) {
+        return true;
+      }
+
+      return getArgument(ctx, (CallExpression) expr, ENABLED)
+        .filter(argEnabledTrace -> !argEnabledTrace.hasExpression(isFalse())).isEmpty();
+    };
+  }
+
+  private static Predicate<Expression> isSensitiveOptionDict(SubscriptionContext ctx) {
+    return expression -> expression.is(Tree.Kind.DICTIONARY_LITERAL)
+      && asDictionaryKeyValuePairs(expression)
+      .filter(isKey(ENABLED))
+      .allMatch(isValueFalse(ctx));
+  }
+
+  private static Stream<KeyValuePair> asDictionaryKeyValuePairs(Expression expression) {
+    return asDictionaryElements(expression)
+      .filter(element -> element.is(Tree.Kind.KEY_VALUE_PAIR)).map(KeyValuePair.class::cast);
+  }
+
+  private static Stream<DictionaryLiteralElement> asDictionaryElements(Expression expression) {
+    return Optional.of(expression)
+      .filter(expr -> expr.is(Tree.Kind.DICTIONARY_LITERAL)).map(DictionaryLiteral.class::cast)
+      .map(DictionaryLiteral::elements)
+      .orElse(Collections.emptyList()).stream();
+  }
+
+  private static Predicate<KeyValuePair> isKey(String keyName) {
+    return keyValuePair -> Optional.of(keyValuePair)
+      .map(KeyValuePair::key)
+      .filter(key -> key.is(Tree.Kind.STRING_LITERAL)).map(StringLiteral.class::cast)
+      .filter(string -> string.trimmedQuotesValue().equals(keyName))
+      .isPresent();
+  }
```

Comment:
Much better!
You miss creating a trace also for the keys and values. They also can be an expression -> variables -> and so their value can be defined somewhere else. Keep it for now. We can discuss an improvement later.